src/variable.jl

```diff
@@ -10,13 +29,19 @@ import .Ops:
 
 import Distributions
 
-type Variable <: AbstractTensor
-    var_node::Operation
-    assign_node::Operation
+type Variable{T} <: tf.AbstractTensor
```

Comment:
I like adding {T} to `Variable`.
Since now all subtypes of `AbstractTensor` have it.
I think we should make it `AbstractTensor{T}`.
That will make a bunch of dispatch easier, eg in indexing.
And also will make it trivial to close #226